src/Components/Notifications/Notification.tsx

```diff
@@ -66,44 +64,39 @@ const Notification: React.FC<NotificationProps> = ({ notificationId }) => {
   useEffect(() => {
     if (
       notification &&
-      !SUPPORTED_NOTIFICATION_TYPES.includes(
-        notification?.notificationType as string
-      )
+      !SUPPORTED_NOTIFICATION_TYPES.includes(notification?.notificationType)
     ) {
-      router.replace(notification?.targetHref as string)
+      router.push(notification?.targetHref as string)
     }
   }, [notification, router])
 
   useEffect(() => {
-    if (!notification) {
-      return
-    }
+    if (!notification) return
 
-    markAsRead()
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [notification])
-
-  const markAsRead = async () => {
-    if (!relayEnvironment || !notification) {
-      return
-    }
-
-    try {
-      const response = await markNotificationAsRead(
-        relayEnvironment,
-        notification.id,
-        notification.internalID
-      )
-      const responseOrError = response.markNotificationAsRead?.responseOrError
-      const errorMessage = responseOrError?.mutationError?.message
+    const markAsRead = async () => {
```

Comment:
shouldn't we use the hook you created here `useMarkNotificationAsRead`?

Reply:
Might as well